util-app/src/main/scala/com/twitter/app/App.scala

```diff
@@ -373,8 +383,13 @@ trait App extends Closable with CloseAwaitably {
     close(defaultCloseGracePeriod)
 
     // The deadline to 'close' is advisory; we enforce it here.
-    if (!suppressGracefulShutdownErrors) Await.result(this, closeDeadline - Time.now)
-    else {
+    if (!suppressGracefulShutdownErrors) {
+      try {
+        Await.result(this, closeDeadline - Time.now)
+      } catch {
+        case t: TimeoutException => throw newCloseException(Seq(t))
+      }
+    } else {
       try { // even if we suppress shutdown errors, we give the resources time to close
         Await.ready(this, closeDeadline - Time.now)
```

Comment:
I'm not sure what should be the behaviour here. Based on the comment on `suppressGracefulShutdownErrors` I would expect the App to still try to kill the JVM in case of error, simply with `System.exit(0)` instead of `System.exit(1)`.